pkg/proof/querier.go

```diff
@@ -123,13 +123,14 @@ func QueryShareInclusionProof(_ sdk.Context, path []string, req abci.RequestQuer
 
 // ParseNamespace validates the share range, checks if it only contains one namespace and returns
 // that namespace ID.
+// The provided range, defined by startShare and endShare, is end-exclusive.
 func ParseNamespace(rawShares []shares.Share, startShare, endShare int) (appns.Namespace, error) {
 	if startShare < 0 {
 		return appns.Namespace{}, fmt.Errorf("start share %d should be positive", startShare)
 	}
 
-	if endShare < 0 {
-		return appns.Namespace{}, fmt.Errorf("end share %d should be positive", endShare)
+	if endShare <= startShare {
+		return appns.Namespace{}, fmt.Errorf("end share %d cannot be lower or equal to the starting share %d", endShare, startShare)
 	}
 
 	if endShare < startShare {
```

Comment:
Couldn't these two if statements be combined together

Reply:
certainly 👍 [7b3e658](https://github.com/celestiaorg/celestia-app/pull/3710/commits/7b3e658bfc2616ff2621ccb85628d644f6b0be9c)